### PR TITLE
[MNT] Add build_tools script to convert deepDNAshape TF weights to .pt

### PR DESCRIPTION
**Pull request by Alleny244**
# PR Description

## Reference Issues/PRs

See also #110 and #729 (deepDNAshape Torch port). This PR adds supporting tooling to convert the remaining upstream TensorFlow checkpoints into PyTorch `.pt` weights used by that work.

---

## What does this implement/fix?

This PR adds `build_tools/convert_deepdnashape.py`, a CLI utility that converts a local directory of deepDNAshape TensorFlow checkpoints (`Feature.index` + `Feature.data-*`) into PyTorch `state_dict` (`.pt`) files compatible with `pyaptamer.deepdnashape._model.DNAModel`.

### Features

- Interactive prompts for input/output directories (or `--models-dir` / `--outdir`)
- Automatically discovers all `*.index` features in the input directory
- Maps TensorFlow variables to PyTorch keys, including:
  - Conv1D kernel axis transposition
  - GRU weight mapping
  - BatchNorm parameter mapping
- Optional `--validate` flag to load each generated `.pt` into `DNAModel`
- Adds a Ruff per-file ignore so `build_tools/*.py` can use `print()` for CLI output

---

## What should a reviewer concentrate their feedback on?

- Correctness of the TensorFlow → PyTorch weight mapping, especially:
  - Conv1D kernel transposition
  - GRU parameter mapping
  - BatchNorm parameter mapping
- Ensuring the conversion tool remains under `build_tools/` and is **not** packaged as part of the library
- CLI usability:
  - Interactive prompts vs. command-line flags
  - Automatic feature discovery from `*.index`
- (Optional) Convert a known feature (for example, `MGW`) and compare the generated `.pt` with an existing checkpoint

---

## Did you add any tests?

No.

This is a one-off maintainer utility used to convert TensorFlow checkpoints into PyTorch weights. It requires both TensorFlow and PyTorch and is not part of the runtime library.

An optional `--validate` flag is provided to verify that each generated checkpoint can be successfully loaded into `DNAModel`.

---

## Additional comments

Upstream currently distributes all pretrained models as TensorFlow checkpoints, while only a subset has been converted to PyTorch and published on Hugging Face.

This utility is intended to generate the remaining `.pt` checkpoints for upload (for example, under the `gcos` feature), independently of the deepDNAshape API PR.

---

## PR Checklist

- [x] The PR title starts with one of:
  - `[ENH]` – New feature or enhancement
  - `[BUG]` – Bug fix
  - `[DOC]` – Documentation improvements
  - `[MNT]` – Maintenance, CI, or test framework updates
- [ ] Added or modified tests (not applicable for this maintainer-only conversion tool)
- [x] Ran `pre-commit` hooks to ensure the code complies with project standards.

### CODE REVIEW

File: build_tools/convert_deepdnashape.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
+#!/usr/bin/env python3
```
Comment (siddharth7113) [inline build_tools/convert_deepdnashape.py]: Not needed.

**Review by siddharth7113 — CHANGES_REQUESTED** [review]
Can you  rename the folder to `scripts/`, also add a README where you have mention about this file and what algorithm this is for, otherwise this is okay since this would be sitting in root folder.

File: build_tools/convert_deepdnashape.py (hunk 1 above)
Comment (Alleny244) [reply build_tools/convert_deepdnashape.py]: Removed

**Comment by Alleny244** [comment]
> Can you rename the folder to `scripts/`, also add a README where you have mention about this file and what algorithm this is for, otherwise this is okay since this would be sitting in root folder.

Done!!

**Review by siddharth7113 — APPROVED** [review]
